src/back/hlsl/writer.rs

```diff
@@ -124,7 +124,8 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                     | crate::MathFunction::Unpack2x16float => {
                         self.need_bake_expressions.insert(arg);
                     }
-                    crate::MathFunction::CountLeadingZeros => {
+                    crate::MathFunction::CountTrailingZeros
+                    | crate::MathFunction::CountLeadingZeros => {
```

Comment:
You only have one instance of `self.write_expr(module, arg, func_ctx)?;` so baking isn't necessary here. This is further explained in https://github.com/gfx-rs/naga/pull/2226#pullrequestreview-1275659048.